apps/src/sites/studio/pages/blocks/edit.js

```diff
@@ -20,50 +20,62 @@ let poolField, nameField, helperEditor, configEditor, validationDiv;
 
 $(document).ready(() => {
   registerReducers({animationList: animationList});
-  getStore().dispatch(setInitialAnimationList(defaultSprites));
-
-  poolField = document.getElementById('block_pool');
-  nameField = document.getElementById('block_name');
-  Blockly.inject(document.getElementById('blockly-container'), {
-    assetUrl,
-    valueTypeTabShapeMap: valueTypeTabShapeMap(Blockly),
-    typeHints: true
-  });
-
-  const blockConfigElement = document.getElementById('block_config');
-
-  // Pretty print the config
-  let blocks = blockConfigElement.value;
-  if (blocks) {
-    blockConfigElement.value = JSON.stringify(JSON.parse(blocks), null, 2);
-  }
-
-  validationDiv = $(
-    blockConfigElement.parentNode.insertBefore(
-      document.createElement('div'),
-      blockConfigElement.nextSibling
-    )
-  );
-
-  const helperCodeElement = document.getElementById('block_helper_code');
-  configEditor = initializeCodeMirror(blockConfigElement, 'application/json', {
-    callback: validateBlockConfig,
-    onUpdateLinting: onUpdateLinting
-  });
-
-  helperEditor = initializeCodeMirror(helperCodeElement, 'javascript', {
-    callback: _ => validateBlockConfig(),
-    onUpdateLinting: onUpdateLinting
-  });
-  poolField.addEventListener('change', updateBlockPreview);
+  getDefaultListMetadata()
+    .then(defaultSprites => {
```

Comment:
nit: now that everything within this `document.ready` callback is another callback, maybe we should pull all of this code into a separate function for readability, like:

```javascript
$(document).ready(() => {
  getDefaultListMetadata()
    .then(someFunc)
    .catch();
});

function someFunc(defaultSprites) {}
```